pkg/pillar/types/patchenvelopestypes.go

```diff
@@ -47,85 +36,31 @@ type PatchEnvelopeInfo struct {
 	VolumeRefs  []BinaryBlobVolumeRef
 }
 
-// since we use json in pubsub we cannot use json - tag
-// and therefore we need some representational structure
-// for PatchEnvelopeInfo.
-type peInfoToDisplay struct {
-	PatchID     string
-	BinaryBlobs []BinaryBlobCompleted
-	VolumeRefs  []BinaryBlobVolumeRef
-}
-
-// PatchEnvelopesJSONForAppInstance returns json representation
-// of Patch Envelopes list which are shown to app instances
-func PatchEnvelopesJSONForAppInstance(pe []PatchEnvelopeInfo) ([]byte, error) {
-	toDisplay := make([]peInfoToDisplay, len(pe))
-
-	for i, envelope := range pe {
-		toDisplay[i] = peInfoToDisplay{
-			PatchID:     envelope.PatchID,
-			BinaryBlobs: envelope.BinaryBlobs,
-			VolumeRefs:  envelope.VolumeRefs,
-		}
-	}
-
-	return json.Marshal(toDisplay)
+// Key for pubsub
+func (pe *PatchEnvelopeInfoList) Key() string {
+	return "zedagent"
```

Comment:
For singleton pubsub messages we ten to use "global" as the key.
"zedagent" is used only for DPC, but in that case there are multiple sources and zedagent is only one of them (i.e. it is not a singleton).

Reply:
Okay, will rename it